agimus_controller/agimus_controller/robot_model_factory/robot_model_factory.py

```diff
@@ -0,0 +1,144 @@
+        if free_flyer:
+            self._model, self._cmodel, self._vmodel = pin.buildModelsFromUrdf(
+                filename=str(urdf),
+                root_joint=pin.JointModelFreeFlyer(),
+                verbose=verbose,
+            )
+        else:
+            self._model, self._cmodel, self._vmodel = pin.buildModelsFromUrdf(
+                filename=str(urdf), root_joint=None, verbose=verbose
+            )
+
+    def _load_default_configuration(self, srdf_path: Path, q0_name: str) -> None:
+        if not srdf_path.is_file():
+            return
+        pin.loadReferenceConfigurations(self._model, str(srdf_path), False)
+        if q0_name and q0_name in self._model.referenceConfigurations:
+            self._q0 = self._model.referenceConfigurations[q0_name]
+        else:
+            self._q0 = pin.neutral(self._model)
+
+    def _load_reduced_model(self, locked_joint_names, q0_name) -> None:
+        locked_joint_ids = [self._model.getJointId(name) for name in locked_joint_names]
+        self._rmodel, geometric_models_reduced = pin.buildReducedModel(
+            self._model,
+            list_of_geom_models=[self._cmodel, self._vmodel],
+            list_of_joints_to_lock=locked_joint_ids,
+            reference_configuration=self._q0,
+        )
+        self._rcmodel, self._rvmodel = geometric_models_reduced
+
+        if q0_name and q0_name in self._rmodel.referenceConfigurations:
+            self._q0 = self._rmodel.referenceConfigurations[q0_name]
+        else:
+            self._q0 = pin.neutral(self._rmodel)
+
+    def _update_collision_model(
+        self,
+        env: Union[Path, None],
+        collision_as_capsule: bool,
+        self_collision: bool,
+        srdf: Path,
+    ) -> None:
+        if collision_as_capsule:
+            self._rcmodel = self._collision_parser.transform_model_into_capsules(
+                self._rcmodel
+            )
+        if self_collision and srdf.exists():
+            self._rcmodel = self._collision_parser.add_self_collision(
+                self._rmodel, self._rcmodel, srdf
+            )
+        if env is not None:
+            self._rcmodel = self._collision_parser.add_collisions(self._rcmodel, env)
+
+    def get_complete_robot_model(self) -> pin.Model:
+        return self._model.copy()
+
+    def get_complete_collision_model(self) -> pin.GeometryModel:
+        return self._cmodel.copy()
+
+    def get_complete_visual_model(self) -> pin.GeometryModel:
+        return self._vmodel.copy()
+
+    def get_reduced_robot_model(self) -> pin.Model:
+        return self._rmodel.copy()
+
+    def get_reduced_collision_model(self) -> pin.GeometryModel:
+        return self._rcmodel.copy()
+
+    def get_reduced_visual_model(self) -> pin.GeometryModel:
+        return self._rvmodel.copy()
+
+    def get_default_configuration(self) -> np.array:
+        return self._q0.copy()
+
+    def get_model_parameters(self) -> RobotModelParameters:
+        return deepcopy(self._params)
```

Comment:
can't we use properties instead of getters here?

Reply:
I changed the API that it actually reflects the usage of a factory:
```python
f = BananaFactory()
a_banana = f.create_banana()
```